Feat/orders date management

Parle648 opened the pull request:
## Describe your changes

- Create logic for calendar at first stage (get dates from orders)

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-46)

### **General**

- [x] types for input and output params
- [x] no `any`, should be also added to eslint rules
- [x] try/catch for any async function
- [x] no **magic** [numbers](/7d77574ee4bc4e339a68066762e887cb)
- [x] compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] avoid similar types with duplicating by generics
- [x] no **console.log** in pr
- [x] .env file should be in .gitignore
- [x] delete commented code if it's not part of documentation
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file
- [x] **constants** and **function** names should be lowercase.
- [x] no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] strings should be in the constant variable. Example: instead of ‘15 3 \* \* _’, const EACH_DAY_15h_15min = ‘15 3 _ \* \*’

### Backend

- [x] swagger for each endpoint. add in the documentation, different types of responses.
      on a successful response (2xx), on unsuccessful response (4xx, 5xx)
- [x] less requests to db, all data should be taken with one query
- [x] **public** in method use only is use function externally
- [x] use ConfigService instead of **process.env**
- [x] use @`@index` decorator for frequently requested data
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] add a set of rules for [nestjs](https://github.com/darraghoriordan/eslint-plugin-nestjs-typed) to the .eslint config file. It helps to prevent bugs and confusing moments
- [x] use [REST API Naming Convention](https://medium.com/@nadinCodeHat/rest-api-naming-conventions-and-best-practices-1c4e781eb6a5) for endpoints

  P.S: more information about RESTful API design by [Microsof](https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design)t

- [x] use ‘**UUIDs**` for primary key, ** to prevent Enumeration Attacks**

Parle648 committed:
+ Create: logic layout

Parle648 committed:
+ Create: date management logic

Parle648 committed:
+ Add: swagger

lyudad commented on src/modules/orders/orders.service.ts:
```diff
@@ -93,4 +93,30 @@ export class OrdersService {
+    } catch (error) {
+      throw new InternalServerErrorException();
+    }
+  }
```
думаю тут логически не совсем правильный подход
если представить что это реальная система. ордеров будет много, доставать все будущие а потом еще и циклом парсить, будет долго по времени. нужно хотя бы ставить верхнюю границу. на ui у нас по месяцам, значит от сегодня до конца месяца. если переключит календарь на следующий месяц то от начала до конца следующего

Parle648 replied:
сделал так что берутся даты в диапазоне 1 месяца а не все orders

Parle648 committed:
* Increace: code quality

Parle648 committed:
* Fix: get dates from one month

lyudad commented on src/modules/orders/orders.service.ts:
```diff
@@ -93,4 +93,28 @@ export class OrdersService {
+
+  async getDates(date: Date, companyId: number): Promise<Record<string, number>> {
+    try {
+      const datePlus30 = new Date(date);
```
why we need date + 30? if todat 21 nov it will return 21 des, but I need to get slot from 21 to 30 nov. this formula work only for next month, as in this case I need from 1st to 31st... if I understand your idea right

Parle648 replied:
In this commit I've changed the getDates logic. Now if the
start date equals today, we get dates from today to 6th of the
next month. In case start date less than today the same
process occurs. However, if the date is later than today
we get dates from 24 previous month to 6th next month, it's
specific of our calendar UI

Parle648 committed:
* Update: get dates logic

In this commit I've changed the getDates logic. Now if the
start date equals today, we get dates from today to 6th of the
next month. In case start date less than today the same
process occurs. However, if the date is later than today
we get dates from 24 previous month to 6th next month, it's
specific of our calendar UI.

Parle648 committed:
- Delete: useless 11 month condition
